packages/template-retail-react-app/app/pages/login/index.jsx

```diff
@@ -170,11 +174,9 @@ const Login = ({initialView = LOGIN_VIEW}) => {
     useEffect(() => {
         if (isRegistered) {
             handleMergeBasket()
-            if (location?.state?.directedFrom) {
-                navigate(location.state.directedFrom)
-            } else {
-                navigate('/account')
-            }
+            const redirectTo = redirectPath ? redirectPath : '/account'
+            navigate(redirectTo)
+            setRedirectPath('')
```

Comment:
I feel like this line here can potentially cause react warning about unsettable state. We are navigating away before the set can be set.
I tried to check out code locally to test out, but I keep running into error with typescript from commerce-sdk-react despite some effects with workarounds.